.github/workflows/test.yml

```diff
@@ -100,7 +100,23 @@ jobs:
           go env -u CC
           go env -u CXX
 
-      # TODO: add Windows 386 tests
+      - name: go test (Windows 386)
+        if: runner.os == 'Windows'
+        run: |
+          env CGO_ENABLED=0 GOARCH=386 go test -shuffle=on -v -count=10 ./...
+          env CGO_ENABLED=1 GOARCH=386 go test -shuffle=on -v -count=10 ./...
+
+      - name: go test (Linux 386)
+        if: runner.os == 'Linux'
+        run: |
+           sudo dpkg --add-architecture i386
+           sudo apt-get update
+           sudo apt-get install ppa-purge
+           sudo ppa-purge -y ppa:ubuntu-toolchain-r/test # Hack for 32bit Linux (#2667)
+           sudo apt-get install gcc-multilib
+           sudo apt-get install g++-multilib
+           sudo apt-get install libasound2-dev:i386 libgl1-mesa-dev:i386 libxcursor-dev:i386 libxi-dev:i386 libxinerama-dev:i386 libxrandr-dev:i386 libxxf86vm-dev:i386
+           env CGO_ENABLED=1 GOARCH=386 go test -shuffle=on -v -count=10 ./...
```

Comment:
I was wondering if your simpler way would work:

```
          go env -w CC=i386-linux-gnu-gcc
          go env -w CXX=i386-linux-gnu-g++
          env CGO_ENABLED=1 GOOS=linux GOARCH=386 go test -shuffle=on -v -count=10 ./...
```

Reply:
Or I prefer

```
env CGO_ENABLED=1 CC=i386-linux-gnu-gcc CXX=i386-linux-gnu-g++ GOOS=linux GOARCH=386 go test -shuffle=on -v -count=10 ./...
```

Reply:
I believe CC and CXX are defaulted correctly by Go. I grab it in the buildShared function for tests and those test did pass on 386. Most of the code is for install the tools which Go doesn't do

Reply:
I mean we might be able to omit apt-get

Reply:
> I believe CC and CXX are defaulted correctly by Go.

Ah ok that's right. 

Can we still need the apt-get lines?